spring-ws-core/src/main/java/org/springframework/ws/client/core/observation/DefaultWebServiceTemplateConvention.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2005-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.ws.client.core.observation;
+
+import io.micrometer.common.KeyValue;
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.Observation;
+
+/**
+ * ObservationConvention that describes how a WebServiceTemplate is observed.
+ * @author Johan Kindgren
+ */
+public class DefaultWebServiceTemplateConvention implements WebServiceTemplateConvention {
+
+    private static final KeyValue EXCEPTION_NONE = KeyValue.of(WebServiceTemplateObservationDocumentation.LowCardinalityKeyNames.EXCEPTION, KeyValue.NONE_VALUE);
+    private String name = "webservice.client";
+
+
+    @Override
+    public KeyValues getLowCardinalityKeyValues(WebServiceTemplateObservationContext context) {
+        return KeyValues.of(
```

Comment:
Can any of these values be null? If that's the case you must not set the key value with a null

Reply:
All of the values are either "unknown" or "none" if no value is present.